Can't package pyusb, 'Unable to find "libusb-1.0.so.0"'

To reproduce

1. `virtualenv env`
`source env/bin/activate`
`pip install https://github.com/pyinstaller/pyinstaller/archive/develop.zip`
`pip install pyusb`
`pip freeze > requirements.txt`
2. Create simple python file which imports pyusb
test.py
```
import usb.core
print 'Success!
```
3. Run the python file normally
```
$ python test.py
Success!
```
4. Package it with pyinstaller
```
pyinstaller --log-level=DEBUG test.py &> log.txt
```
log.txt
```
14 INFO: PyInstaller: 3.3.dev0+5fd013bfe
14 INFO: Python: 2.7.12+
14 INFO: Platform: Linux-4.8.0-53-generic-x86_64-with-Ubuntu-16.10-yakkety
14 INFO: wrote /home/joel/Code/random/pyinstaller/test.spec
15 DEBUG: Testing for UPX ...
17 INFO: UPX is not available.
17 DEBUG: script: /home/joel/Code/random/pyinstaller/test.py
17 INFO: Extending PYTHONPATH with paths
['/home/joel/Code/random/pyinstaller', '/home/joel/Code/random/pyinstaller']
17 INFO: checking Analysis
17 INFO: Building Analysis because out00-Analysis.toc is non existent
17 INFO: Initializing module dependency graph...
18 INFO: Initializing module graph hooks...
19 DEBUG: Hidden import: codecs
65 DEBUG: Hidden import 'codecs' already found
65 INFO: running Analysis out00-Analysis.toc
65 DEBUG: Analyzing /home/joel/Code/random/pyinstaller/env/bin/python
71 DEBUG: Skipping libdl.so.2 dependency of python
71 DEBUG: Skipping libc.so.6 dependency of python
71 DEBUG: Skipping libutil.so.1 dependency of python
71 DEBUG: Skipping libpthread.so.0 dependency of python
71 DEBUG: Skipping libm.so.6 dependency of python
72 DEBUG: Adding libz.so.1 dependency of python from /lib/x86_64-linux-gnu/libz.so.1
72 DEBUG: Analyzing /lib/x86_64-linux-gnu/libz.so.1
78 DEBUG: Skipping libc.so.6 dependency of libz.so.1
78 INFO: Caching module hooks...
81 INFO: Analyzing /home/joel/Code/random/pyinstaller/test.py
1078 INFO: Loading module hooks...
1079 INFO: Loading module hook "hook-usb.py"...
Unable to find "libusb-1.0.so.0" when adding binary and data files.
```
This causes the packaging process to halt with an empty `dist/` and `build/test/` folders.

I dug around and found out, that `hook-usb.py` reports the binaries to be `[('libusb-1.0.so.0', '')]` and I think this causes that  `src_root_path_or_glob`(`utils.py`, lines 450+) to just contain 'libusb1.0.so.0', which produces just `[]`after it get's run through the `glob.glob()` and not the real path, which I think is`/lib/x86_64-linux-gnu/libusb-1.0.so.0` on my computer.

Environment:
```
Python 2.7.12+
Ubuntu 16.10
Virtualenv 15.1.0
```
requirements.txt
```
appdirs==1.4.3
packaging==16.8
PyInstaller==3.3.dev0+5fd013bfe
pyparsing==2.2.0
pyusb==1.0.0
six==1.10.0
```

Also, this seems to be somewhat related to #1682 and commit https://github.com/pyinstaller/pyinstaller/commit/23901eb1ce7e60f1aac2a73e6cb3b02f1a34b9b1

I'm seeing some [build failures](https://travis-ci.org/pyinstaller/pyinstaller/builds/337087283) that don't look like they're related to my work. Python 2.7 Linux, 3.4 Linux, 3.6 Mac, and 3.6 Linux all pass.

Maybe a core team member or maintainer could help me understand if these are related, and how to move forward?

Python 3.3 Linux:
```
$ py.test -n 3 --maxfail 3 tests/functional/test_libraries.py tests/functional/test_hooks
usage: py.test [options] [file_or_dir] [file_or_dir] [...]
py.test: error: unrecognized arguments: -n tests/functional/test_libraries.py tests/functional/test_hooks
  inifile: None
  rootdir: /home/travis/build/pyinstaller/pyinstaller
The command "py.test -n 3 --maxfail 3 tests/functional/test_libraries.py tests/functional/test_hooks" exited with 2.
```

Python 3.5 Linux:
```
=================================== FAILURES ===================================
________________________ test_pandas_extension[onefile] ________________________
[gw0] linux -- Python 3.5.4 /home/travis/virtualenv/python3.5.4/bin/python
pyi_builder = <conftest.AppBuilder object at 0x7fd4643e2ef0>
    @importorskip('pandas')
    def test_pandas_extension(pyi_builder):
        # Tests that C extension 'pandas.lib' is properly bundled. Issue #1580.
        pyi_builder.test_source(
            """
            from pandas.lib import is_float
            assert is_float(1) == 0
>           """)
E       Failed: Running exe /tmp/pytest-of-travis/pytest-0/popen-gw0/test_pandas_extension_onefile_0/dist/test_pandas_extension failed with return-code 1.
/home/travis/build/pyinstaller/pyinstaller/tests/functional/test_libraries.py:806: Failed
----------------------------- Captured stdout call -----------------------------
------- Starting build. -------
Syntax error in  /home/travis/virtualenv/python3.5.4/lib/python3.5/site-packages/jinja2/asyncsupport.py
("'yield' inside async function", ('/home/travis/virtualenv/python3.5.4/lib/python3.5/site-packages/jinja2/asyncsupport.py', 35, 12, '            yield event\n'))
------- Build finshed, now running executable. -------
RUNNING:  '/tmp/pytest-of-travis/pytest-0/popen-gw0/test_pandas_extension_onefile_0/dist/test_pandas_extension' , args:  ['./test_pandas_extension']
TIMED OUT:  '/tmp/pytest-of-travis/pytest-0/popen-gw0/test_pandas_extension_onefile_0/dist/test_pandas_extension' , args:  ['./test_pandas_extension']
----------------------------- Captured stderr call -----------------------------
```

--- a/PyInstaller/hooks/hook-usb.py
+++ b/PyInstaller/hooks/hook-usb.py
@@ -47,6 +47,7 @@ try:
 except (ValueError, usb.core.USBError) as exc:
     logger.warning("%s", exc)
 
+
 # if nothing found, try to use our custom mechanism
 if not binaries:
     # Try to resolve your libusb libraries in the following order:

--- a/PyInstaller/hooks/hook-usb.py
+++ b/PyInstaller/hooks/hook-usb.py
@@ -40,11 +40,12 @@ try:
     binaries = []
     for usblib in [getattr(usb.backend, be)._lib for be in backends]:
         if usblib is not None:
-            binaries = [(usblib._name, '')]
+            binaries = _resolveCtypesImports([os.path.basename(usblib._name)])
+            assert len(binaries[0]) == 3
+            binaries = [(binaries[0][1], '')]
 
 except (ValueError, usb.core.USBError) as exc:
     logger.warning("%s", exc)
-
 
 # if nothing found, try to use our custom mechanism
 if not binaries:

